alibi/explainers/integrated_gradients.py

```diff
@@ -722,6 +773,8 @@ def __init__(self,
         self._is_np: Optional[bool] = None
         self.orig_dummy_input: Optional[Union[list, np.ndarray]] = None
 
+        self.target_fn = target_fn
```

Comment:
I personally do not want to add more dummy calls,

but in this case we have the keras model and target_fn if set, we could check that this is ok at init time?

I guess you left it to explain time to have the check in one place but perhaps if we check `target_fn` this is towards checking the parameters passed and providing some hints back to the use if things are good.

Reply:
I think this is quite hard without not knowing what the shape of the inputs is which we only know at `explain` time. If we wanted to deeply validate user models and callables at construction time it would involve a sizable refactor as we would also need to ask the user for the type of data that can be put through the model.

Practically, this could be solved by asking a user to provide a sample batch of data at `__init__` that we can use to validate the model and any callables. In theory this could be an optional `kwarg` so that people would only use it when they want some guarantees about the functionality of the explainer post `__init__` (we wouldn't want to make it mandatory, I think).

I think this should be discussed in a different place (maybe a new issue?). Would be keen to hear thoughts from @ascillitoe and @arnaudvl.

Reply:
Agree with the sentiments of @sakoush and @jklaise here; as @sakoush hints at, any checks we can do to enable more helpful error messages are great. But as @jklaise alludes to, the complexity needed for this quickly ramps up if we don't assume the inputs pass a certain sanity threshold to begin with. This is made even more challenging by the number of ways tensorflow and pytorch models can fail!

Providing a sample batch of data could be a nice solution, but it might be a little unintuitive. If we were going to go down this route, perhaps it should be a well-documented optional functionality? i.e. have an option to provide sample data so that `target_fn` can be checked at init.

Reply:
If we have the keras model + layer, we can get the shape of the input data shape ((i.e. `model.get_layer(layer).input_shape`)) and therefore able to construct a valid dummy? 

This could also help constructing target_fn if we just refer to `argmax` as example computation.

but yeah out of scope of this PR perhaps.

Reply:
If I remember correctly, I think the `input_shape` being present is limited to a subset of `tensorflow` models as the model may be in a state where `input_shape` is unknown (there are some hints of this in the `integrated_gradients.py` module already). More generally, thinking ahead, for `torch` models input shapes are generally unknown unless provided by the user.